File: app/src/main/java/net/albertogarrido/studydrivetest/Item.java
package net.albertogarrido.studydrivetest;

import org.threeten.bp.Instant;
import org.threeten.bp.LocalDateTime;
import org.threeten.bp.ZoneId;
import org.threeten.bp.format.DateTimeFormatter;
import org.threeten.bp.format.FormatStyle;

public final class Item {

    private String id;
    private String date;
    private int color;

    public Item() {
        Instant instant = Instant.now();
        ZoneId zoneId = ZoneId.of("Europe/Berlin");
        LocalDateTime localDateTime = LocalDateTime.ofInstant(instant, zoneId);
        id = String.valueOf(instant.getEpochSecond());
        date = localDateTime.format(DateTimeFormatter.ofLocalizedDateTime(FormatStyle.MEDIUM));
        color = Utils.getRandomColor();
    }

    public String getId() {
        return id;
    }

    public String getDate() {
        return date;
    }

    public int getColor() {
        return color;
    }

}

File: app/src/main/java/net/albertogarrido/studydrivetest/Item.kt
package net.albertogarrido.studydrivetest

import org.threeten.bp.Instant
import org.threeten.bp.LocalDateTime
import org.threeten.bp.ZoneId
import org.threeten.bp.format.DateTimeFormatter
import org.threeten.bp.format.FormatStyle

class Item(
        val id: String,
        val date: String,
        val color: Int
) {
    companion object {
        @JvmStatic
        fun create(): Item {
            val instant = Instant.now()
            val zoneId = ZoneId.of("Europe/Berlin")
            val localDateTime = LocalDateTime.ofInstant(instant, zoneId)
            return Item(
                    id = instant.epochSecond.toString(),
                    date = localDateTime.format(DateTimeFormatter.ofLocalizedDateTime(FormatStyle.MEDIUM)),
                    color = randomColor()
            )
        }
    }

}
